video/external/src/decoder/openh264.rs

```diff
@@ -67,7 +73,136 @@ pub struct H264Decoder {
+                "libopenh264-2.4.1-mac-arm64.dylib",
+                "213ff93831cfa3dd6d7ad0c3a3403a6ceedf4ac1341e1278b5b869d42fefb496",
+            ),
+            ("windows", "x86") => (
+                "openh264-2.4.1-win32.dll",
+                "83270149640469c994a62cc32a6d8c0413cd7b802b7f1f2f532159f5bdc1cedd",
+            ),
+            ("windows", "x86_64") => (
+                "openh264-2.4.1-win64.dll",
+                "081b0c081480d177cbfddfbc90b1613640e702f875897b30d8de195cde73dd34",
+            ),
+            (os, arch) => return Err(format!("Unsupported OS/arch: {}/{}", os, arch).into()),
+        };
+
+        Ok(OpenH264Data {
+            local_filenames,
+            download_filename,
+            download_sha256,
+        })
+    }
+
+    fn download_openh264(
+        openh264_data: &OpenH264Data,
+        directory: &Path,
+    ) -> Result<PathBuf, Box<dyn std::error::Error>> {
+        // See the license at: https://www.openh264.org/BINARY_LICENSE.txt
+        const URL_BASE: &str = "http://ciscobinary.openh264.org/";
+        const URL_SUFFIX: &str = ".bz2";
+
+        let (filename, sha256sum) = (
+            openh264_data.download_filename,
+            openh264_data.download_sha256,
+        );
+
+        std::fs::create_dir_all(directory)?;
+        let filepath = directory.join(filename);
+
+        // If the binary doesn't exist in the expected location, download it.
+        if !filepath.is_file() {
+            let url = format!("{}{}{}", URL_BASE, filename, URL_SUFFIX);
+            let response = reqwest::blocking::get(url)?;
+            let mut bzip2_reader = BzDecoder::new(response);
+
+            let mut tempfile = tempfile::NamedTempFile::with_prefix_in(filename, directory)?;
+            copy(&mut bzip2_reader, &mut tempfile)?;
+            // Let's assume that if this fails, it's because another process has already put it there
+            // and loaded it, therefore it can't be overwritten (on Windows at least), but in the end,
+            // all's fine - the hash will still be checked before attempting to load the library.
+            let _ = tempfile.persist(&filepath);
+        }
+
+        // Regardless of whether the library was already there, or we just downloaded it, let's check the MD5 hash.
+        let mut sha256 = Sha256::new();
+        copy(&mut File::open(filepath.clone())?, &mut sha256)?;
+        let sha256digest = sha256.finalize();
+        let result: [u8; 32] = sha256digest.into();
+
+        if result[..] != hex::decode(sha256sum)?[..] {
+            let size = filepath.metadata().map(|f| f.len()).unwrap_or_default();
+            return Err(format!(
+                "SHA256 checksum mismatch for {filename}; expected {sha256sum}, found {sha256digest:x} (with a size of {size} bytes)",
+            )
+            .into());
+        }
+
+        Ok(filepath)
+    }
+
+    pub fn load_openh264(directory: &Path) -> Result<OpenH264Codec, Box<dyn std::error::Error>> {
```

Comment:
I believe these methods would fit better in `OpenH264Codec`, this one for instance would be a constructor

Reply:
Good point, moved them there. I still need to bikeshed the names and their order. Any suggestions?

Reply:
IMO names are okay (`get_openh264_data`, `download_openh264`, `load_openh264`, only one of them is pub anyways), I don't see how we can improve here. The current order is bottom-up, you could reverse it to obtain top-down order, but I'm not sure if we have any preference here.

Reply:
> IMO names are okay `get_openh264_data`, `download_openh264`, `load_openh264`

The current names are not these - but since they are within `OpenH264Codec`, and as you said, mostly not `pub`, do you still think they should contain `openh264`?

Reply:
Nope, they don't have to :) it's all good